src/processor/tasks.py

```diff
@@ -60,3 +62,52 @@ def pull_metrics():
 
     scrape.end_date = datetime.utcnow()
     db.session.commit()
+
+
+@celery_app.task(name='check-wikipedia-references')
+def check_wikipedia_references():
+    """ Check all Wikipedia events to see that DOIs are still referenced."""
+
+    for event in Event.query.filter_by(
+            origin=Origins.wikipedia.value,
+            is_deleted=False
+    ):
+
+        if not check_wikipedia_event(event):
+            db.session.add(
+                RawEvent(
+                    event=event,
+                    origin=Origins.wikipedia.value,
+                    provider=StaticProviders.hirmeos_altmetrics.value,
+                    created_at=datetime.utcnow(),
+                    reason_for_deletion='Entry DOI is no longer referenced on '
```

Comment:
`reason_for_deletion` is indeed a good idea, but we might turn this into a Reason object with some attributes (e.g. `Reason.doi_not_on_wikipedia_page`) with a DB value that is easy enough to query later (e.g. `doi-not-in-wiki-page`).